torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1228,7 +1227,7 @@ def make_singular_matrix_batch_base(size, rank):
             u_diag_abs_largest = u_diag_abs.max(dim=-1, keepdim=True).values
             u_diag_abs_smallest_idxs = torch.topk(u_diag_abs, k=(n - rank), largest=False).indices
             u.diagonal(0, -2, -1).div_(u_diag_abs_largest)
-            u.diagonal(0, -2, -1)[..., u_diag_abs_smallest_idxs] = torch.finfo(dtype).eps
+            u[..., u_diag_abs_smallest_idxs] = torch.finfo(dtype).eps
```

Comment:
This change is fine and does not break anything? I wonder how a 1D index gets properly understood in the 2D "diagonal" context...

Reply:
the test was failing and I had to change it. Here all I'm doing is making the whole column(s) to be epsilon hence effectively reducing the norm of the matrix, which is something desirable in this sampling..